applications/jonny5/src/j5_balance_crawler.erl

```diff
@@ -0,0 +1,242 @@
+        'true' -> gen_server:start_link(?MODULE, [], []);
+        'false' -> 'ignore'
+    end.
+
+%%====================================================================
+%% gen_server callbacks
+%%====================================================================
+
+%%--------------------------------------------------------------------
+%% Function: init(Args) -> {ok, State} |
+%%                         {ok, State, Timeout} |
+%%                         ignore               |
+%%                         {stop, Reason}
+%% Description: Initiates the server
+%%--------------------------------------------------------------------
+init([]) ->
+    wh_util:put_callid(?MODULE),
+    sleep_and_cast(0,'crawl_accounts'),
+    {'ok', []}.
+
+%%--------------------------------------------------------------------
+%% Function: %% handle_call(Request, From, State) -> {reply, Reply, State} |
+%%                                      {reply, Reply, State, Timeout} |
+%%                                      {noreply, State} |
+%%                                      {noreply, State, Timeout} |
+%%                                      {stop, Reason, Reply, State} |
+%%                                      {stop, Reason, State}
+%% Description: Handling call messages
+%%--------------------------------------------------------------------
+handle_call(_Request, _From, State) ->
+    Reply = {'error', 'not_implemented'},
+    {'reply', Reply, State}.
+
+%%--------------------------------------------------------------------
+%% Function: handle_cast(Msg, State) -> {noreply, State} |
+%%                                      {noreply, State, Timeout} |
+%%                                      {stop, Reason, State}
+%% Description: Handling cast messages
+%%--------------------------------------------------------------------
+handle_cast('crawl_accounts', []) ->
+    Cycle = whapps_config:get_integer(?APP_NAME, <<"balance_crawler_cycle_ms">>, ?DEFAULT_CRAWLER_CYCLE),
+    spawn_sleep_and_cast(Cycle, 'crawl_accounts'),
+    case j5_channels:accounts() of
+        [] ->
+            {'noreply', [], 'hibernate'};
+        Accounts ->
+            lager:debug("check ~p account(s) for zero balance", [length(Accounts)]),
+            sleep_and_cast(0, 'next_account'),
+            {'noreply', Accounts}
+    end;
+
+%%--------------------------------------------------------------------------------------------
+%% If crawler still checking account from previous cycle, warn and try again after some delay
+%%--------------------------------------------------------------------------------------------
+handle_cast('crawl_accounts', [_|_] = Accounts) ->
+    lager:warning("delay next crawler cycle for ~p ms, ~p account(s) to go. Please tune crawler settings", [?OVERLOAD_DELAY, length(Accounts)]),
+    spawn_sleep_and_cast(?OVERLOAD_DELAY, 'crawl_accounts'),
+    {'noreply', Accounts};
+
+handle_cast('next_account', []) ->
+    {'noreply', [], 'hibernate'};
```

Comment:
Why not `spawn_sleep_and_cast/2` here and forget about the `overload_delay` stuff above?